main/tray.js

```diff
@@ -85,6 +91,40 @@ function setupIpcEventListeners (contextMenu) {
     getItemById('checkingForUpdates').visible = false
   })
 
+  ipcMain.on(ipcMainEvents.ACTIVITY_LOGGED, () => {
+    assert(tray)
+
+    let connected = false
+    for (const entry of ctx.getAllActivities().reverse()) {
+      if (
+        entry.source === 'Saturn' &&
+        entry.type === 'info' &&
+        entry.message.includes('Saturn Node is online')
+      ) {
+        connected = true
+        break
+      } else if (
+        entry.source === 'Saturn' &&
+        (
+          entry.type === 'error' ||
+          (
+            entry.message === 'Saturn Node started.' ||
+            entry.message.includes('was able to connect') ||
+            entry.message.includes('will try to connect')
+          )
+        )
+      ) {
+        connected = false
+        break
+      }
+    }
```

Comment:
I find it a bit hacky and potentially brittle to detect online/offline mode here in the tray module. Have you considered moving this code into the saturn module?

For example:
- The saturn module (and each new module in the future) should keep track whether it's online or offline, potentially using this activity-log-based solution. The status can be exposed as a property, e.g. `saturn.isOnline` or maybe `saturn.status` that can be `'online' | 'offline' | 'connecting'`.
- Here in the tray, we can keep listening for ACTIVITY_LOGGEd events and use the `saturn.status` variable to decide whether we are online or not. In the future, when we add more modules, we can extend this check to combine status from all modules.

Reply:
good idea to consolidate the concerns, will do!